ext/ftp/php_ftp.c

```diff
@@ -76,7 +104,16 @@ PHP_MINIT_FUNCTION(ftp)
 #endif
 #endif
 
-	le_ftpbuf = zend_register_list_destructors_ex(ftp_destructor_ftpbuf, NULL, le_ftpbuf_name, module_number);
+	INIT_CLASS_ENTRY(ce, "FTPConnection", NULL);
+	php_ftp_ce = zend_register_internal_class(&ce);
+	php_ftp_ce->ce_flags |= ZEND_ACC_FINAL;
```

Comment:
The other migrated objects also have the  `ZEND_ACC_NO_DYNAMIC_PROPERTIES` flag

Reply:
Thanks. Added.  FWIW HashContext does NOT have that set. We should probably add it.